pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository.
+	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
+
+	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
+	Jenkinsfile string `json:"jenkinsfile,omitempty"`
+
+	// Env contains additional environment variables you want to pass to the Jenkins Pipeline.
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
Actually I'm wondering if its better to define environments at the namespace/project level instead. Its common for teams to have multiple apps they work on and they often want to build/test/stage/release them together. i.e. if you work on 3 apps, stage them all in the same namespace by default.

So rather than each BuildConfig having its own set of environments; I'm thinking the better default is to do it at the project level.  Then as folks create new apps, they already have their environments setup. 

If a team works on apps A and B and they want A to have its own staging environment separate from B etc; then just move A and B into separate projects etc.

So the more common case is more like, a team has a 'source project' where they create their apps and the BuildConfigs; then that project can be associated with other environments for testing, staging, production etc. Then the team only needs to setup its environments once and then they can just create/update appps